cli/cmd/cluster_kubeconfig.go

```diff
@@ -27,20 +27,62 @@ func (r *runners) InitClusterKubeconfig(parent *cobra.Command) *cobra.Command {
 		Long:         `Download credentials for a test cluster`,
 		RunE:         r.kubeconfigCluster,
 		SilenceUsage: true,
-		Args:         cobra.MinimumNArgs(1),
 	}
 	parent.AddCommand(cmd)
 
+	cmd.Flags().StringVar(&r.args.kubeconfigClusterName, "name", "", "name of the cluster to download credentials for (when id is not provided)")
+	cmd.Flags().StringVar(&r.args.kubeconfigClusterID, "id", "", "id of the cluster to download credentials for (when name is not provided)")
+	cmd.Flags().StringVar(&r.args.kubeconfigPath, "output", "", "path to kubeconfig file to write to, if not provided, it will be merged into your existing kubeconfig")
+
 	return cmd
 }
 
 func (r *runners) kubeconfigCluster(_ *cobra.Command, args []string) error {
 	kotsRestClient := kotsclient.VendorV3Client{HTTPClient: *r.platformAPI}
 
-	kubeconfig, err := kotsRestClient.GetClusterKubeconfig(args[0])
+	// by default, we look at args[0] as the id
+	// but if it's not provided, we look for a viper flag named "name" and use it
+	// as the name of the cluster, not the id
+	clusterID := ""
+	if len(args) > 0 {
+		clusterID = args[0]
+	} else if r.args.kubeconfigClusterName != "" {
+		clusters, err := kotsRestClient.ListClusters(false)
+		if err != nil {
+			return errors.Wrap(err, "list clusters")
+		}
+		for _, cluster := range clusters {
+			if cluster.Name == r.args.kubeconfigClusterName {
+				clusterID = cluster.ID
+				break
+			}
```

Comment:
There can be multiple clusters with the same name.
Ok to always take the first one. But I think in other cases (for example channel commands), we return an error that we can not uniquely define it.